### PR TITLE
Enable Javadoc checks for `src/test` directories

### PR DESCRIPTION
**Pull request by Ao-senXiong**
See discussion in https://github.com/eisop/checker-framework/pull/827#discussion_r1722425425

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -633,6 +633,7 @@ task allJavadoc(type: Javadoc, group: 'Documentation') {
             project(':checker-qual').sourceSets.main.allJava,
             project(':checker').sourceSets.main.allJava,
             project(':framework').sourceSets.main.allJava,
+            project(':framework').sourceSets.test.allJava,
```
Comment (wmdietl) [inline build.gradle:636]: This is great and reveals 14 errors in javadoc!
Should the same be done for `:checker`?

Comment (Ao-senXiong) [reply build.gradle:636]: Sounds good to me! I will udpate the gradle file.

Comment (wmdietl) [reply build.gradle:636]: There is also `dataflow/src/test`, `checker/src/testannotations`, and `checker-util/src/test`.
Please also fix the javadoc errors revealed by the new check.

Comment (Ao-senXiong) [reply build.gradle:636]: Okay, sounds good.

Comment (Ao-senXiong) [reply build.gradle:636]: Maybe framework-test/src/test as well?

Comment (Ao-senXiong) [reply build.gradle:638]: I will leave do `checker/src/testannotations` later in a new PR. Looks like we need to define the source set for testannotations folder.

**Review by wmdietl — APPROVED** [review]
Thanks for all the cleanups!

**Pull request merged**